### ISSUE TITLE
[BUG] unknown node type: desc_returns

### ISSUE THREAD

**Issue by haukex** [liran-funaro/sphinx-markdown-builder#19]
**Describe the bug**
When using `autodoc` to generate the documentation for a module, for a function with a return type, the warning `docstring of ...: WARNING: unknown node type: <desc_returns: ...>` occurs and the return type is not part of the output. For example: https://gist.github.com/haukex/2b012e1490b538210d7279b94bffc4d5#file-foo-py

**To Reproduce**
1. `git clone https://gist.github.com/2b012e1490b538210d7279b94bffc4d5.git bug_desc_returns`
2. `cd bug_desc_returns`
3. `pip install -r requirements.txt`
4. `make`
5. Output includes `.../bug_desc_returns/foo.py:docstring of foo.foo:1: WARNING: unknown node type: <desc_returns: <#text: 'int'>>`
6. `cat output/markdown/index.md`:
   ```
   <a id="module-foo"></a>
   
   Hello, World!
   
   ### foo.foo(bar: str)
   
   Foo!
   ```
   Note how the return type `int` is missing.

**Expected behavior**
I would expect the warning not to occur and the function signature to include `-> int`.

**Environment**
Python 3.12 on Ubuntu Linux

**Additional context**
I'm not sure, but it seems to me like [sphinx_markdown_builder/translator.py](https://github.com/liran-funaro/sphinx-markdown-builder/blob/main/sphinx_markdown_builder/translator.py) needs to add support for [sphinx.addnodes.desc_returns](https://www.sphinx-doc.org/en/master/extdev/nodes.html#sphinx.addnodes.desc_returns)?

**Comment by liran-funaro** [comment]
@haukex  Thank you for the report.
I'm sorry it took so much time to reply.
I believe your observation is correct. I'll be sure to take care of this issue in the following weeks.

**Comment by liran-funaro** [comment]
@haukex I can't reproduce this issue.
For me it produces the correct result.

I run the following procedure (Python 3.12):
```shell
python3 -m venv ./venv-issue-19
. venv-issue-19/bin/activate
pip3 install sphinx==7.2.6
pip3 install .[dev]
make test
```

**Comment by haukex** [comment]
Hi, thanks for the reply and for looking into this. The test code you provided does seem to work on my system, however, the test case I provided above still produces the same problem. The difference appears to be this setting in your `conf.py`: `autodoc_typehints = "description"`, putting this in my test causes the issue to go away, and removing this setting from your tests causes the issue to appear there. This means that the problem appears to be with the [default](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_typehints) `autodoc_typehints = "signature"`. Is this supposed to be supported by `sphinx-markdown-builder`?

**Comment by liran-funaro** [comment]
@haukex I've added support for `autodoc_typehints=signature`.

You can install the latest commit and see if it solves this issue for you.
If it solves it, I'll upload a new version soon.

```shell
pip uninstall sphinx-markdown-builder
pip install git+https://github.com/liran-funaro/sphinx-markdown-builder.git@main
```

**Comment by haukex** [comment]
That works, thank you very much! Output of my test is now:
```markdown
<a id="module-foo"></a>

Hello, World!

### foo.foo(bar: str) → int

Foo!
```

**Comment by liran-funaro** [comment]
Great. I'll release a new version soon.

**Comment by liran-funaro** [comment]
This issue should be resolved in the latest release ([0.6.7](https://pypi.org/project/sphinx-markdown-builder/0.6.7/)). Please verify that it works for you, and close this issue if it does.
